calabash/Classes/Utils/LPJSONUtils+Introspection.m

```diff
@@ -0,0 +1,146 @@
+      NSString *propName        = objCStr(property_getName(property));
+      NSDictionary *accessors   = [self customAccessorsForProperty:property name:propName object:object];
+      customAccessors[@"properties"][propName] = accessors;
+    }
+    free(properties);
+    
+    /* Get selectors for current class */
+    Method *methods = class_copyMethodList(c, &methodCount);
+    for (int i = 0; i < methodCount; i++) {
+      SEL method = method_getName(methods[i]);
+      NSString *methodName = NSStringFromSelector(method);
+      if ([object respondsToSelector:method]) {
+        [customAccessors[@"methods"] addObject:methodName];
+      } else {
+        NSLog(@"Object doesn't respond to method: ");
+      }
+    }
+    free(methods);
+    
+  } while (( c = [c superclass] ));
+  
+  return customAccessors;
+}
+
+/*
+ customAccessorsForProperty
+ Returns custom named accessors, if any, for a given property of an object.
+ 
+ @param property   the property for which you'd like the custom accessors
+ @return  NSString the custom accessors if any (formatted like a JSON object),
+ or nil if no custom properties exist.
+ 
+ See https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/ObjCRuntimeGuide/Articles/ocrtPropertyIntrospection.html
+ */
++ (NSDictionary *)customAccessorsForProperty:(objc_property_t)property name:(NSString *)propName object:(id)obj {
+  unsigned int attributeCount;
+  NSMutableDictionary *accessors = [NSMutableDictionary dictionaryWithCapacity:2];
+  
+  BOOL  customGetter  = NO,
+        customSetter  = NO,
+        readonly      = NO;
+  
+  objc_property_attribute_t *attributes = property_copyAttributeList(property, &attributeCount);
+  for (int j = 0; j < attributeCount; j++) {
+    objc_property_attribute_t attribute = attributes[j];
+      
+    NSString *value       = objCStr(attribute.value);
+    switch (attribute.name[0]) {
+        
+      case 'G': /* custom getter */
+        customGetter = YES;
+        accessors[@"getter"] = value;
+        break;
+        
+      case 'S': /* custom setter */
+        customSetter = YES;
+        accessors[@"setter"] = value;
+        break;
+        
+      case 'R': /* readonly */
+        readonly = YES;
+        accessors[@"setter"] = @"READONLY";
+        break;
+        
+      default:
+        break;
+    }
+  }
+  if (!customGetter && [obj respondsToSelector:NSSelectorFromString(propName)]) {
+    accessors[@"getter"] = propName;
+  }
+  if (!customSetter && !readonly) {
+    NSString *firstLetterCap  = [[propName substringToIndex:1] uppercaseString];
+    NSString *theRest         = [propName substringFromIndex:1];
+    NSString *setter          = [NSString stringWithFormat:@"set%@%@:", firstLetterCap, theRest];
+    if ([obj respondsToSelector:NSSelectorFromString(setter)]) { accessors[@"setter"] = setter; }
+    else { accessors[@"setter"] = @"<UNKNOWN>"; }
+  }
+  free(attributes);
+  return accessors;
+}
+
+#define BUF_LEN 256
```

Comment:
I am trying to move away from `#define`, especially those that do not appear at the top of a file.

Does it make sense to make this a static const?

```
static NSUInteger const LPJSONUtilsIntrospectionMethodDescriptionBufferLength = 256;
```